### PR TITLE
[OPPRO-291] Support hash(),get_json_object() functions in Gluten

### PR DESCRIPTION
**Pull request by zhixingheyi-tian**
## What changes were proposed in this pull request?

Support hash(),get_json_object() functions in Gluten

## How was this patch tested?

UT added

### CODE REVIEW

File: jvm/src/main/scala/io/glutenproject/expression/MultaryExpressionTransformer.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import io.glutenproject.expression.ConverterUtils.FunctionConfig
+import io.glutenproject.substrait.expression.{ExpressionBuilder, ExpressionNode}
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.types.DataType
+
+import scala.collection.mutable.ArrayBuffer
+
+class Murmur3HashTransformer(exps: Seq[Expression], original: Expression)
```
Comment (jinchengchenghh) [inline jvm/src/main/scala/io/glutenproject/expression/MultaryExpressionTransformer.scala:28]: Can you rename this file to HashExpressionTransformer, there will be many hash expressions

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala [hunk 2]
```diff
@@ -290,6 +290,67 @@ class TestOperator extends WholeStageTransformerSuite {
     assert(df.queryExecution.executedPlan.find(_.isInstanceOf[ProjectExecTransformer]).isDefined)
   }
 
+  test("Test round function") {
+    val df = spark.sql("SELECT round(cast(l_orderkey as int), 2) from lineitem limit 1")
+    val result = df.collect()
+    assert(result.length == 1)
+    val expected = Seq(Row(1))
+    TestUtils.compareAnswers(result, expected)
+    df.show()
```
Comment (jinchengchenghh) [inline backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala:299]: Does this line can be removed?

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala [hunk 3]
```diff
@@ -290,6 +290,67 @@ class TestOperator extends WholeStageTransformerSuite {
...
+    df.explain(false)
+    df.printSchema()
```
Comment (jinchengchenghh) [inline backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala:301]: as above

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala [hunk 4]
```diff
@@ -290,6 +290,67 @@ class TestOperator extends WholeStageTransformerSuite {
...
+    assert(result.length == 1)
+    val expected = Seq(Row(-1712319331))
+    TestUtils.compareAnswers(result, expected)
+    df.show()
+    df.explain(false)
+    df.printSchema()
+    assert(df.queryExecution.executedPlan.find(_.isInstanceOf[ProjectExecTransformer]).isDefined)
+  }
+
+  test("Test get_json_object function") {
```
Comment (PHILO-HE) [inline backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala:341]: Do we really need to add such test by using tpc-h table. The table is not actually used for get_json_object and we are just doing the test with literal input.

**Comment by PHILO-HE** [comment]
For `get_json_object`, a few UTs have been added in our arrow/gandiva repo, some of which were extracted from customer's real-world workload. I think we can port these test cases into gluten. The spark UT for `get_json_object` does not covering many of these cases.
See code link:
https://github.com/oap-project/arrow/blob/arrow-4.0.0-oap/cpp/src/gandiva/json_holder_test.cc

File: jvm/src/main/scala/io/glutenproject/expression/MultaryExpressionTransformer.scala (hunk 1 above)
Comment (zhixingheyi-tian) [reply jvm/src/main/scala/io/glutenproject/expression/MultaryExpressionTransformer.scala]: HashExpression belongs to MultaryExpression.
Later, other functions will use new design implementation.

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala (hunk 2 above)
Comment (zhixingheyi-tian) [reply backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala]: Now, have use new Suite.

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala (hunk 3 above)
Comment (zhixingheyi-tian) [reply backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala]: ditto

File: backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala (hunk 4 above)
Comment (zhixingheyi-tian) [reply backends-velox/src/test/scala/io/glutenproject/execution/TestOperator.scala]: Don't have to use tpc-h table.
Have created VeloxFunctionsValidateSuite as new functions UTs.

**Comment by zhixingheyi-tian** [comment]
> For `get_json_object`, a few UTs have been added in our arrow/gandiva repo, some of which were extracted from customer's real-world workload. I think we can port these test cases into gluten. The spark UT for `get_json_object` does not covering many of these cases. See code link: https://github.com/oap-project/arrow/blob/arrow-4.0.0-oap/cpp/src/gandiva/json_holder_test.cc

How about put these UTs in oap-project/velox later?

**Review by zhouyuan — APPROVED** [review]

**Pull request merged**